packages/vite/src/node/plugins/terser.ts

```diff
@@ -2,18 +2,29 @@ import type { Plugin } from '../plugin'
 import { Worker } from 'okie'
 import type { Terser } from 'types/terser'
 import type { ResolvedConfig } from '..'
+import { requireResolveFromRootWithFallback } from '../utils'
+
+let terserPath: string | undefined
+const loadTerserPath = (root: string) => {
+  if (terserPath) return terserPath
+  try {
+    terserPath = requireResolveFromRootWithFallback(root, 'terser')
+  } catch (e) {
+    if (e.code === 'MODULE_NOT_FOUND') {
+      throw new Error(`terser not found. Did you install it?`)
+    } else {
+      const message = new Error(`terser failed to load:\n${e.message}`)
+      message.stack = e.stack + '\n' + message.stack
+      throw message
+    }
+  }
+  return terserPath
+}
 
 export function terserPlugin(config: ResolvedConfig): Plugin {
   const makeWorker = () =>
     new Worker(
-      (basedir: string, code: string, options: Terser.MinifyOptions) => {
-        // when vite is linked, the worker thread won't share the same resolve
-        // root with vite itself, so we have to pass in the basedir and resolve
-        // terser first.
-        // eslint-disable-next-line node/no-restricted-require
-        const terserPath = require.resolve('terser', {
-          paths: [basedir]
-        })
+      (terserPath: string, code: string, options: Terser.MinifyOptions) => {
         return require(terserPath).minify(code, options) as Terser.MinifyOutput
```

Comment:
I'm not sure if this part of my change is correct.
The test is passing though I don't quite understand this code.